oshdb-filter/src/main/java/org/heigit/ohsome/oshdb/filter/ChangesetIdFilterEquals.java

```diff
@@ -0,0 +1,49 @@
+package org.heigit.ohsome.oshdb.filter;
+
+import org.heigit.ohsome.oshdb.osh.OSHEntity;
+import org.heigit.ohsome.oshdb.osm.OSMEntity;
+import org.heigit.ohsome.oshdb.util.mappable.OSMContribution;
+import org.heigit.ohsome.oshdb.util.mappable.OSMEntitySnapshot;
+import org.jetbrains.annotations.Contract;
+
+/**
+ * A filter which selects OSM contributions by a changeset id.
+ */
+public class ChangesetIdFilterEquals extends NegatableFilter {
```

Comment:
`NegatableFilter` blindly negates all `FilterExpression` methods including `applyOSM` but which needs always and intentional for contributionTypeFilters return true, because of possible minor-versions. 
As a suggestion here [see](https://github.com/GIScience/oshdb/blob/ad8b0abc5997617fb4ffe444b8264e9cbc458c36/oshdb-filter/src/main/java/org/heigit/ohsome/oshdb/filter/ContributionFilter.java#L98) which we can discuss in PR #385

As an example in pseudo code
```
var osh = OSHWay (
  OSMWay ( changeset:42
    OSHNode(
      OSMNode(changeset:12))))

var expression = parser.parse("not changeset:42")
expression.applyOSH(osh) -> true (node changeset:12)
osh.getVersions().foreach(osm -> expression.applyOSM(osm)) -> always false
expression.applyOSMContribution(contrib(osh)) -> true
```